libs/execution/src/lib/blocks/block-execution-util.ts

```diff
@@ -0,0 +1,107 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import {
+  BlockDefinition,
+  collectParents,
+} from '@jvalue/jayvee-language-server';
+
+import { ExecutionContext } from '../execution-context';
+import { Logger } from '../logger';
+import { IOTypeImplementation, NONE } from '../types';
+
+// eslint-disable-next-line import/no-cycle
+import { createBlockExecutor } from './block-executor-registry';
+import * as R from './execution-result';
+
+export interface ExecutionOrderItem {
+  block: BlockDefinition;
+  value: IOTypeImplementation | null;
+}
+
+export async function executeBlocks(
+  executionContext: ExecutionContext,
+  executionOrder: ExecutionOrderItem[],
+  initialInputValue: IOTypeImplementation | undefined = undefined,
```

Comment:
I don't entirely understand when this value is set. I assume when executing a composite block?

Maybe a good indicator to add JSDoc comment above explaining the parameter.